modules/vpc/variables.tf

```
@@ -1,85 +1,152 @@
+    condition     = var.mtu >= 1300 && var.mtu <= 8896
+    error_message = "MTU Range must be between 1300 and 8896 !"
+  }
 }
 
-variable "region" {
+variable "routing_mode" {
   description = <<-EOF
-  GCP region for all the created subnetworks and for all the imported subnetworks. Set to null to use a default provider's region.
-  
-  To add subnetworks with another region use a separate instance of this module (and specify `create_network=false` to avoid creating a duplicate network).
+  Type of network-wide routing mode to use. Possible types are: REGIONAL and GLOBAL.
+  REGIONAL routing mode will set the cloud routers to only advertise subnetworks within the same region as the router.
+  GLOBAL routing mode will set the cloud routers to advertise all the subnetworks that belong to this network.
   EOF
-  default     = null
+  default     = "REGIONAL"
   type        = string
+  validation {
+    condition     = var.routing_mode == "REGIONAL" || var.routing_mode == "GLOBAL"
+    error_message = "Routing mode must be either 'REGIONAL' or 'GLOBAL'."
+  }
 }
 
-variable "allowed_protocol" {
-  description = "A protocol (TCP or UDP) to pass for the `networks` entries that do not have their own `allowed_protocol` attribute."
-  default     = "all"
-}
+variable "firewall_rules" {
+  description = <<-EOF
+  A map containing each firewall rule configuration.
+  Action of the firewall rule is always `allow`.
+  The only possible direction of the firewall rule is `INGRESS`.
 
-variable "allowed_ports" {
-  description = "A list of ports to pass for the `networks` entries that do not have their own `allowed_ports` attribute. For example [\"22\", \"443\"]. Can also include ranges, for example [\"80\", \"8080-8999\"]. Empty list means to allow all."
-  default     = []
-  type        = list(string)
+  List of available attributes of each firewall rule entry:
+  - `name` : Name of the firewall rule.
+  - `source_ranges` : (Optional) A list of strings containing the source IP ranges to be allowed on the firewall rule.
+  - `source_tags` : (Optional) A list of strings containing the source network tags to be allowed on the firewall rule.
+  - `source_service_accounts` : (Optional) A list of strings containg the source servce accounts to be allowed on the firewall rule.
+  - `target_service_accounts` : (Optional) A list of strings containing the service accounts for which the firewall rule applies to.
+  - `target_tags` : (Optional) A list of strings containing the network tags for which the firewall rule applies to. 
+  - `allowed_protocol` : The protocol type to match in the firewall rule. Possible values are : tcp, udp, icmp, esp, ah, sctp, ipip, all
+  - `ports` : A list of strings containing TCP or UDP port numbers to match in the firewall rule. This type of setting can be configured if allowing only TCP and UDP as protocols.
+  - `priority` : (Optional) A priority value for the firewall rule. The lower the number - the more preffered the rule is.
```

```suggestion
  - `priority` : (Optional) A priority value for the firewall rule. The lower the number - the more preferred the rule is.
```
Or "[...] the higher the priority of the rule."